### PR TITLE
[NCL-6854] Update auto-merge workflow

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

**Comment by DnsZhou** [comment]
Checked the code and verified on local workflow env(with nektos/act tool).
image

**Review by DnsZhou — APPROVED** [review]

**Review by matedo1 — CHANGES_REQUESTED** [review]

File: .github/workflows/auto-merge.yml [hunk 1]
```diff
@@ -11,18 +11,34 @@ jobs:
...
       - name: Dependabot metadata
-        id: metadata
-        uses: dependabot/fetch-metadata@v1.1.1
+        id: dependabot-metadata
+        uses: dependabot/fetch-metadata@v1.3.5
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
       - name: Approve a PR
-        run: gh pr review --approve "$PR_URL"
+        run: gh pr review --approve "$PR_URL" -b "Pull request **approved** because it is a patch or minor update"
```
[the remark this answers is not in the document]
Comment (patrikk0123) [reply .github/workflows/auto-merge.yml:19]: Deleted this anyway.

File: .github/workflows/auto-merge.yml [hunk 2]
```diff
@@ -11,18 +11,34 @@ jobs:
...
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: Automerge Dependabot PRs for patch and minor updates
+      - name: Enable auto-merge (rebase) for a PR for patch and minor update starting with caret
         if: ${{startsWith(steps.dependabot-metadata.outputs.previous-version, '^') && (steps.metadata.outputs.update-type == 'version-update:semver-patch' || steps.metadata.outputs.update-type == 'version-update:semver-minor')}}
         run: gh pr merge --auto --rebase "$PR_URL"
         env:
           PR_URL: ${{github.event.pull_request.html_url}}
           GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+      - name: Comment a PR for patch and minor update not starting with caret
+        if: ${{!startsWith(steps.dependabot-metadata.outputs.previous-version, '^') && (steps.metadata.outputs.update-type == 'version-update:semver-patch' || steps.metadata.outputs.update-type == 'version-update:semver-minor')}}
```
[the remark this answers is not in the document]
Comment (patrikk0123) [reply .github/workflows/auto-merge.yml]: Yes

File: .github/workflows/auto-merge.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrikk0123) [reply .github/workflows/auto-merge.yml]: Yes, it was moved to correct place.

**Review by matedo1 — APPROVED** [review]

**Pull request merged**